src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Presentation/SampleChooserViewModel.cs

```diff
@@ -248,7 +248,7 @@ public bool Matches(string path)
 
 			private bool Matches(string category, string sampleName)
 			{
-				return category.HasValue() &&
```

Comment:
`HasValue` and `HasValueTrimmed` are redundant both have the same implementation. The original intent is that `HasValue` matches `!IsNullOrEmpty` and `HasValueTrimmed` matches `!IsNullOrWhiteSpace`.

I removed `HasValue` and `HasValueTrimmed` completely, since it is easier to understand the `!IsNullOrEmpty` and `!IsNullOrWhiteSpace` at callsites.